src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -270,6 +293,37 @@ private void styleFetchedListView(CheckListView<CitationRelationItem> listView)
         listView.setSelectionModel(new NoSelectionModel<>());
     }
 
+    private EntryEditor createEntryEditor() {
+        Supplier<LibraryTab> tabSupplier = () -> this.libraryTab;
+        return new EntryEditor(this.libraryTab,
+                // Actions are recreated here since this avoids passing more parameters and the amount of additional memory consumption is neglegtable.
+                new UndoAction(tabSupplier, dialogService, stateManager),
+                new RedoAction(tabSupplier, dialogService, stateManager));
+    }
+
+    private void showEntrySourceDialog(CodeArea codeArea) {
+        if (codeArea == null) {
+            dialogService.showWarningDialogAndWait(Localization.lang("BibTeX source", databaseContext.getMode().getFormattedName()), Localization.lang("Could not load %0 source", databaseContext.getMode().getFormattedName()));
```

Comment:
How often does this occur? Bug in the code or real use? I think, its bug in the code. Thus, please use `LOGGER.warn(...)` instead of a dialog service.

Reply:
Also just return then. #FailFast. Do not use `} else { `.